packages/app/src/utils/dynamicUI/extractDynamicConfig.ts

```diff
@@ -22,6 +24,26 @@ export type MenuItem =
       };
     };
 
+export type ConfigMenuItems = {
+  [key: string]: MenuItemObj;
+};
+
+export type MenuItemObj = {
+  title?: string; // needed for parent menu items
+  icon?: string; // needed for parent menu items
+  parent?: string; // needed for child menu items
+  priority?: number; // needed for both
+};
+
+export type GenericMenuItem = {
```

Comment:
What is the difference between `ConfigMenuItems`, `MenuItemObj` and `GenericMenuItem`?

I guess we should

1. rename the old `MenuItem` to `DynamicRouteMenuItem` 
2. and then have a new `MenuItem` (which is currently called `GenericMenuItem` for the new configuration.

What else is needed?